src/manifolds/GeneralUnitaryMatrices.jl

```diff
@@ -0,0 +1,712 @@
+i.e. that ``p`` has an determinante of absolute value one
+
+The tolerance for the last test can be set using the `kwargs...`.
+"""
+function check_point(
+    M::GeneralUnitaryMatrices{n,𝔽,AbsoluteDeterminantOneMatrices},
+    p;
+    kwargs...,
+) where {n,𝔽}
+    if !isapprox(abs(det(p)), 1; kwargs...)
+        return DomainError(
+            det(p),
+            "The absolute value of the determinant of $p has to be 1 but it is $(abs(det(p)))",
+        )
+    end
+    if !isapprox(p' * p, one(p); kwargs...)
+        return DomainError(
+            norm(p' * p - one(p)),
+            "$p must be orthogonal but it's not at kwargs $kwargs",
+        )
+    end
+    return nothing
+end
+
+@doc raw"""
+    check_point(M::Rotations, p; kwargs...)
+
+Check whether `p` is a valid point on the [`UnitaryMatrices`](@ref) `M`,
+i.e. that ``p`` has an determinante of absolute value one, i.e. that ``p^{\mathrm{H}}p``
+
+The tolerance for the last test can be set using the `kwargs...`.
+"""
+function check_point(
+    M::GeneralUnitaryMatrices{n,𝔽,DeterminantOneMatrices},
+    p;
+    kwargs...,
+) where {n,𝔽}
+    if !isapprox(det(p), 1; kwargs...)
+        return DomainError(det(p), "The determinant of $p has to be +1 but it is $(det(p))")
+    end
+    if !isapprox(p' * p, one(p); kwargs...)
+        return DomainError(
+            norm(p' * p - one(p)),
+            "$p must be orthogonal but it's not at kwargs $kwargs",
+        )
+    end
+    return nothing
+end
+
+function check_size(::GeneralUnitaryMatrices{n}, p) where {n}
+    m = size(p)
+    if length(m) != 2
+        return DomainError(
+            size(p),
+            "The point $p is not a matrix (expected a length of size to be 2, got $(length(size(p))))",
+        )
+    end
+    if m != (n, n)
+        return DomainError(
+            size(p),
+            "The point $p is not a matrix of size $((n,n)), but $(size(p)).",
+        )
+    end
+    return nothing
+end
+function check_size(::GeneralUnitaryMatrices{n}, p, X) where {n}
+    m = size(X)
+    if length(size(X)) != 2
+        return DomainError(
+            size(X),
+            "The tangent vector $X is not a matrix (expected a length of size to be 2, got $(length(size(X))))",
+        )
+    end
+    if m != (n, n)
+        return DomainError(
+            size(X),
+            "The tangent vector $X is not a matrix of size $((n,n)), but $(size(X)).",
+        )
+    end
+    return nothing
+end
+
+@doc raw"""
+    check_vector(M::UnitaryMatrices{n}, p, X; kwargs... )
+    check_vector(M::OrthogonalMatrices{n}, p, X; kwargs... )
+    check_vector(M::Rotations{n}, p, X; kwargs... )
+    check_vector(M::GeneralUnitaryMatrices{n,𝔽}, p, X; kwargs... )
+
+Check whether `X` is a tangent vector to `p` on the [`UnitaryMatrices`](@ref)
+space `M`, i.e. after [`check_point`](@ref)`(M,p)`, `X` has to be skew symmetric (hermitian)
+dimension and orthogonal to `p`.
```

Comment:
"dimension" should probably be deleted here?

Reply:
ah, yes
```suggestion
and orthogonal to `p`.
```